cmake/tribits/common_tpls/FindTPLHDF5.cmake

```diff
@@ -2,78 +2,103 @@
 # See associated tribits/Copyright.txt file for copyright and license! #
 ########################################################################
 
-#
-# First, set up the variables for the (backward-compatible) TriBITS way of
-# finding HDF5.  These are used in case find_package(HDF5 ...) is not called
-# or does not find HDF5.  Also, these variables need to be non-null in order
-# to trigger the right behavior in the function
-# tribits_tpl_find_include_dirs_and_libraries().
-#
-
-set(REQUIRED_HEADERS hdf5.h)
-set(REQUIRED_LIBS_NAMES hdf5)
-
-if (HDF5_REQUIRE_FORTRAN)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} hdf5_fortran)
-endif()
+set(HDF5_INTERNAL_IS_MODERN FALSE)
 
-if (TPL_ENABLE_MPI)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} z)
-endif()
+if (Netcdf_ALLOW_MODERN)
+
+  set(minimum_modern_HDF5_version 1.13.2)
+  print_var(Netcdf_ALLOW_MODERN)
+  message("-- Using find_package(HDF5 ${minimum_modern_HDF5_version} CONFIG) ...")
+  find_package(HDF5  ${minimum_modern_HDF5_version}  CONFIG)
```

Comment:
@ibaned, @gsjaardema pointed out in https://github.com/sandialabs/seacas/pull/336#pullrequestreview-1157178993 that this should really be a minimum version, not a specific version of HDF5.  I don't know if `HDF5Config.cmake` supports version ranges, but if it does, I think this would be:

```
    find_package(HDF5 ${minimum_modern_HDF5_version}... CONFIG)
```

Reply:
I think it was concluded in other discussion that the code as is does request this version or higher. That is certainly my intention anyway, that this work for later versions.